### PR TITLE
Added TaskExecutionThinController and TaskExecutionThinResource.

### PR DESCRIPTION
**Pull request by corneil**

### CODE REVIEW

**Review by cppwfs — CHANGES_REQUESTED** [review]
LGTM.  Just my nagging nitipicks :-)

**Review by onobc — APPROVED** [review]
LGTM - once @cppwfs comments are addressed. Thanks @corneil

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/TaskExecutionsDocumentation.java [hunk 1]
```diff
@@ -263,6 +263,39 @@ public void listTaskExecutions() throws Exception {
...
+					.param("properties", "app.my-task.foo=bar,deployer.my-task.something-else=3")
+					.param("arguments", "--server.port=8080 --foo=bar")
+			)
+			.andExpect(status().isCreated()));
+
+		this.mockMvc.perform(
+				get("/tasks/thinexecutions")
+					.param("page", "1")
+					.param("size", "2"))
+			.andDo(print())
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/TaskExecutionsDocumentation.java:280]: All the other tests have `print()`

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java [hunk 2]
```diff
@@ -113,6 +119,7 @@ public class TaskTemplate implements TaskOperations {
 		Assert.notNull(restTemplate, "RestTemplate must not be null");
 		Assert.isTrue(resources.getLink("about").isPresent(), "Expected about relation");
 		Assert.isTrue(resources.getLink(EXECUTIONS_RELATION).isPresent(), "Executions relation is required");
+		Assert.isTrue(resources.getLink(THIN_EXECUTIONS_RELATION).isPresent(), "Executions relation is required");
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:122]: Rewritten for a list and a more specific message.

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/TaskExecutionsDocumentation.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cppwfs) [reply spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/TaskExecutionsDocumentation.java:280]: True.   But we can remove it from the others unless we are doing diagnostics.    Else it fills the logs.   Its an annoying nitpick :-)

**Comment by cppwfs** [comment]
Rebased Squashed Merged.   Thanks @corneil for the PR!

**Pull request closed without merging**